addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyHubHandler.java

```diff
@@ -124,14 +124,15 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void initialize() {
+        buttonExecutor = Executors.newSingleThreadScheduledExecutor();
         cancelRetry();
         connect();
     }
 
     @Override
     public void dispose() {
-        listeners.clear();
         buttonExecutor.shutdownNow();
```

Comment:
> Note that dispose is not some kind of destructor and was as far as I know never intended like that.

I don't think this is completely correct. As the javadoc also says: "Disposes the thing handler, e.g. deallocate resources.". This indeed is the only place where the ThingHandler can clean up after himself. So I would agree it is the right point in time (and also the only one...) to shutdown the executor. 
To make it look more symmetric it would be nice to create the executor in the initialize() method evem.